hydromt/models/model_api.py

```diff
@@ -32,6 +32,8 @@ class Model(object, metaclass=ABCMeta):
     _GEOMS = {"<general_hydromt_name>": "<model_name>"}
     _MAPS = {"<general_hydromt_name>": "<model_name>"}
     _FOLDERS = [""]
+    # tell hydroMT which methods should receive the res and region arguments
+    _CLI_ARGS = {"region": "setup_basemaps", "res": "setup_basemaps"}
```

Comment:
I think this is good to avoid too much backward compatibility with existing plugins like wflow or sfincs etc.
But if we make a release with this change anyway, shouldn't we try to move away from setup_basemaps in the model api and here use setup_region instead ? res can be removed or stay with setup_basemaps for gridded models. I guess we may get new CLI args when the new model type come in (network, mesh).

Maybe in the end keep like this for now and discuss again during the design network/mesh sessions

Reply:
I indeed used "setup_basemaps" here to make sure all plugins keep running for now. We can decide again later to change this if that makes sense then.

Reply:
the "res" keyword is optional btw. If it's not in this dictionary it will not be used in any method.